### ISSUE TITLE
Emulator not working with multiple express servers on multiple http triggers across different codebases with --inspect-functions 

### ISSUE THREAD

**Issue by EthanSK** [firebase/firebase-tools#8067]
### [REQUIRED] Environment info

**firebase-tools:** 13.29.1

**Platform:** macOS

### [REQUIRED] Test case

have 2 of these in 2 codebases
import { environment } from '@ai-music-video-studio/worker-config';
import { NestFactory } from '@nestjs/core';
import { ExpressAdapter } from '@nestjs/platform-express';
import express from 'express';
import * as admin from 'firebase-admin';
import { onRequest } from 'firebase-functions/https';
import { AppModule } from './app/app.module';
admin.initializeApp();

if (environment.currentEnvironment === 'development') {
  console.log('Using firestore emulator.');
  process.env.FIRESTORE_EMULATOR_HOST = 'localhost:8080';
}

const server = express(); //doesnt work with multiple servers...

const createNestServer = async (expressInstance: express.Express) => {
  console.log('Creating Worker Nest Server');

  const app = await NestFactory.create(
    AppModule,
    new ExpressAdapter(expressInstance),
  );

  await app.init();
};

// createNestServer(server)
//   .then(() => Logger.log('Nest listening'))
//   .catch((err) => Logger.error('Nest Error', err));

// export const worker = onRequest(
//   { timeoutSeconds: 3600, memory: '8GiB' },
//   server,
// );

export const worker = onRequest(
  { timeoutSeconds: 3600, memory: '8GiB' },
  async (req, res) => {
    await createNestServer(server);
    server(req, res);
  },
);

### [REQUIRED] Steps to reproduce

run with --inspect-functions (also using firestore emulator with import and export data)

### [REQUIRED] Expected behavior

 functions to work with debugger

### [REQUIRED] Actual behavior

doesn't work at all, seems to get stuck or something. one of the http servers sometimes works with debugging, but the other one doesnt work at all (the first one sends a request to the second one over http)

**Comment by aalej** [comment]
Hey @EthanSK, thanks for reaching out. I'm working on reproducing this issue, but I'm having some difficulty. To get a better overview of how your functions are set up, and to help us accurately reproduce this error. Could you verify the ff:
- From what I can gather, you have two different endpoints(for example, `api` and `worker`, which exist in their respective codebases). The `api` endpoint makes a request to the `worker` endpoint. Is this assumption correct?

- Does the issue occur when calling the `api` endpoint? If you call the `worker` endpoint, does debugger work or does it also get stuck? 

Also, are there any errors or warnings raised in the `firebase-debug.log` you can share with us?  Let me know in case I may have misunderstood or missed anything here.

**Comment by EthanSK** [comment]
yes the assumption is correct, but its not necessary to reproduce the bug. 

The issue is hard to reproduce in the same way reliably, its just general buggy behaviour. sometimes the debugger works for one of the apps, sometimes not at all, and usually at least on of the apps just doesn't reply to http calls. But it always has at least one of these issues, making the workflow unusable.

**Comment by EthanSK** [comment]
[hs_err_pid3494.log](https://github.com/user-attachments/files/18200892/hs_err_pid3494.log)
also getting errors like this when running multiple codebases even without inspect-functinos

**Comment by aalej** [comment]
Thanks for the additional information. At the moment, I think the issue here might be because when passing the `--inspect-funcitons` flag, all the functions run in a single process. Since it looks like your functions are running in parallel, with some functions relying on other functions to run, this might be causing the emulator to stall other functions while another function is still running. I'll check this with our engineering team.

Regarding the error from your [latest comment](https://github.com/firebase/firebase-tools/issues/8067#issuecomment-2555073007). Could you please open up a new report for this as this seems to be a different issue? Looks related to the Firestore emulator.

**Comment by EthanSK** [comment]
ok, looking forward to the fix, debugging is crucial, and having this multi-codebase setup with proper debugging support can really make firebase a true all in one solution that trumps all else :)

**Comment by EthanSK** [comment]
any updates?

**Comment by joehan** [comment]
Hey @EthanSK - as mentioned above, `--inspect-functions` executes functions in serial, and we don't currently have plans to support parallel execution with it. Unfortunately, it's technically very difficult to support this - in parallel mode, we spin up and shut down lots of separate Node processes, and so it is difficult to keep the your debugger attached to the 'right'  process. Normally, this 

Unfortunately, I think the best way forward for now will be to avoid the `--inspect-functions` flag when you need multiple functions to execute in parallel.
